design-proposals/emf-profiling-analysis.md

```diff
@@ -0,0 +1,872 @@
+    y-axis "CPU Used (%)"
+    bar "CPU Used" [22.6, 6, 15, 23.8]
+```
+
+**Memory Usage (%)**
+
+```mermaid
+xychart-beta
+    title "Memory Usage (%)"
+    x-axis "Profile" ["Baseline-ObsON", "Baseline-ObsOFF", "OnPrem-Azure", "OnPrem-Proxmox"]
+    y-axis "Memory Used (%)"
+    bar "Memory Used" [54.66, 24.48, 81.05, 27.55]
+```
+
+**Disk Usage (%)**
+
+```mermaid
+xychart-beta
+    title "Disk Usage (%)"
+    x-axis "Profile" ["Baseline-ObsON", "Baseline-ObsOFF", "OnPrem-Azure", "OnPrem-Proxmox"]
+    y-axis "Disk Used (%)"
+    bar "Disk Used" [33.24, 22.89, 38.05, 15.26]
+```
+
+Top processes consuming CPU and memory
+
+Baseline profile `dev-internal-coder-autocert.yaml` with `enableObservability` set to `true`:
+
+```sh
+COMMAND         %CPU
+otelcol-contrib 47.7
+kubelet         24.3
+argocd-applicat 20.6
+kube-apiserver  18.8
+mimir            7.3
+containerd       7.0 
+```
+
+```sh
+COMMAND MEM(MiB)
+kube-apiserver       1796.7
+prometheus           1194.4
+java                  797.8
+mimir                 691.4
+mimir                 690.0
+argocd-applicat       658.2
+```
+
+profile for OnPrem deployed on Azure VM:
+
+In case of Azure deployment the topology is deploying VM inside Azure VM that run EMF.
+
+profile for OnPrem deployed on Proxmox VM:
+
+```sh
+COMMAND         %CPU
+kubelet         69.3
+otelcol-contrib 68.5
+argocd-applicat 52.3
+kube-apiserver  35.4
+mimir           18.4
+prometheus      14.6 
+```
+
+```sh
+COMMAND MEM(MiB)            
+kube-apiserver       2041.4 
+prometheus           1535.8 
+java                  861.3 
+argocd-applicat       829.9 
+mimir                 808.4 
+mimir                 779.7 
+```
+
+### Timing analysis
+
+The timing analysis was done using the kubernetes events captured during the deployment of the EMF. It is recommended to run the following commands within an hour of deploying the EMF as the retention policy might overwrite the events captured during the deployment.
+
+```bash
+mkdir -p kind-diagnostics
+kubectl get pods -o wide -A | tee kind-diagnostics/pods-list.txt
+kubectl describe pods -A | tee kind-diagnostics/pods-describe.txt
+mage logutils:collectArgoDiags | tee kind-diagnostics/argo-diag.txt
+kubectl get applications -o yaml -A | tee kind-diagnostics/argocd-applications.yaml
+kubectl get events -o yaml -A | tee kind-diagnostics/events.yaml
+```
+
+Following data was generated for the Coder deployment with `dev-internal-coder-autocert.yaml` and [OnPrem deployment on Azure VM](https://github.com/open-edge-platform/edge-manage-docs/blob/main/docs/developer_guide/set_up_dev_env/index.rst)
+
+#### Argo applications Durations `dev-internal-coder-autocert.yaml`
+
+| Section | Start | End | Duration (sec) [min] |
+|---------|---------------------|---------------------|:----------------------:|
+| root-app | 2025-06-09 11:15:43 | 2025-06-09 11:39:31 | **1428 [23.8 min]** 🔴 |
+| postgresql-secrets | 2025-06-09 11:17:23 | 2025-06-09 11:38:47 | **1284 [21.4 min]** 🔴 |
```

Comment:
secrets-config and psql-secrets so slow is very surprising to me. There is something triggerring changes - psql secrets definetely should be ready at the very beignning and same for secrets-config. For MT components I have no idea but this is suspicious too worth investigating.

Even if we exclude these outliers - we can maybe optimize 1m. Perhaps we should see if we can otpimize the sync waves or in general the sync logic from Argo

Reply:
@pierventre it's not fully true that psql-secrets and secrets-config are slow. The duration here represents the total time from the first event till the last event for that app. In between, the app gets to the synced state multiple time. @krishnajs maybe we should clarify this.